{
    "CVE_data_meta": {
        "ASSIGNER": "secalert@redhat.com",
        "ID": "CVE-2018-16890",
        "STATE": "PUBLIC"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "curl",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "7.64.0"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "The curl Project"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "libcurl versions from 7.36.0 to before 7.64.0 is vulnerable to a heap buffer out-of-bounds read. The function handling incoming NTLM type-2 messages (`lib/vauth/ntlm.c:ntlm_decode_type2_target`) does not validate incoming data correctly and is subject to an integer overflow vulnerability. Using that overflow, a malicious or broken NTLM server could trick libcurl to accept a bad length + offset combination that would lead to a buffer read out-of-bounds."
            }
        ]
    },
    "impact": {
        "cvss": [
            [
                {
                    "vectorString": "5.4/CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:N/A:L",
                    "version": "3.0"
                }
            ]
        ]
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-125"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "name": "DSA-4386",
                "refsource": "DEBIAN",
                "url": "https://www.debian.org/security/2019/dsa-4386"
            },
            {
                "name": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-16890",
                "refsource": "CONFIRM",
                "url": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-16890"
            },
            {
                "name": "https://curl.haxx.se/docs/CVE-2018-16890.html",
                "refsource": "MISC",
                "url": "https://curl.haxx.se/docs/CVE-2018-16890.html"
            },
            {
                "name": "USN-3882-1",
                "refsource": "UBUNTU",
                "url": "https://usn.ubuntu.com/3882-1/"
            },
            {
                "name": "106947",
                "refsource": "BID",
                "url": "http://www.securityfocus.com/bid/106947"
            }
        ]
    }
}